src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
+        toDelete.push_back(childPerformanceTraceId); // We can't delete directly the childPerformanceTraceId from the map as we
+                                                     // are iterating over it
+    }
+    for (auto childPerformanceTraceId2: toDelete) {
+        stopPTrace(childPerformanceTraceId2, aborted);
+    }
+
+    // Stop the PerformanceTrace
+    if (performanceTrace.isSpan()) {
+        if (!performanceTrace.span()) {
+            assert(false && "Span is not valid");
+            return;
+        }
+        if (aborted) sentry_span_set_status(performanceTrace.span(), SENTRY_SPAN_STATUS_ABORTED);
+        sentry_span_finish(performanceTrace.span()); // Automatically stop any child transaction
+    } else {
+        if (!performanceTrace.transaction()) {
+            assert(false && "Transaction is not valid");
+            return;
+        }
+        if (aborted)
+            sentry_transaction_set_status(performanceTrace.transaction(),
+                                          SENTRY_SPAN_STATUS_ABORTED /*This enum is also valid for a transactrion.*/);
+        sentry_transaction_finish(performanceTrace.transaction()); // Automatically stop any child transaction
+    }
+    _performanceTraces.erase(id);
+}
+
+pTraceId Handler::startTransaction(const std::string &name, const std::string &description) {
+    sentry_transaction_context_t *tx_ctx = sentry_transaction_context_new(name.c_str(), description.c_str());
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx, sentry_value_new_null());
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
+    if (!res) {
+        assert(false && "Transaction already exists");
+        return 0;
+    }
+
+    PerformanceTrace &pTrace = it->second;
+    pTrace.setTransaction(tx);
+
+    return _pTraceIdCounter;
+}
+
+pTraceId Handler::startSpan(const std::string &name, const std::string &description, const pTraceId &parentId) {
+    auto parentIt = _performanceTraces.find(parentId);
+    if (parentIt == _performanceTraces.end()) {
+        assert(false && "Parent transaction/span does not exist");
+        return 0;
+    }
+
+    PerformanceTrace &parent = parentIt->second;
+    sentry_span_t *span = nullptr;
+    if (parent.isSpan()) {
+        if (!parent.span()) {
+            assert(false && "Parent span is not valid");
+            return 0;
+        }
+        span = sentry_span_start_child(parent.span(), name.c_str(), description.c_str());
+        assert(span);
+    } else {
+        if (!parent.transaction()) {
+            assert(false && "Parent transaction is not valid");
+            return 0;
+        }
+        span = sentry_transaction_start_child(parent.transaction(), name.c_str(), description.c_str());
+        assert(span);
+    }
+
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
```

Comment:
Same remark.

Reply:
Same:
The second argument of try_emplace is not the value itself but the argument that will be passed to the constructor of PerformanceTrace.

```cpp
namespace Sentry {
class Handler {
     ...
    private:
    std::map<pTraceId, PerformanceTrace> _performanceTraces;
    ....
}
```